lib/curl/curl.nit

```diff
@@ -72,26 +74,18 @@ end
 # CURL HTTP Request
 class CurlHTTPRequest
 	super CurlRequest
-	super CCurlCallbacks
-	super CurlCallbacksRegisterIntern
+	super NativeCurlCallbacks
 
 	var url: String
 	var datas: nullable HeaderMap = null is writable
 	var headers: nullable HeaderMap = null is writable
+	var delegate: nullable CurlCallbacks = null is writable
 
 	# Set the user agent for all following HTTP requests
-	fun user_agent=(name: String)
-	do
-		curl.prim_curl.easy_setopt(new CURLOption.user_agent, name)
-	end
-
-	init (url: String, curl: nullable Curl) is old_style_init do
-		self.url = url
-		self.curl = curl
-	end
+	var user_agent: nullable String = null is writable
```

Comment:
With the new optional attributes, is it still necessary to have it start at null ? It could be an optional part of the constructor if you leave it without the default value